crates/revm/src/db.rs

```diff
@@ -17,14 +17,14 @@ use auto_impl::auto_impl;
 #[auto_impl(& mut, Box)]
 pub trait Database {
     /// Get basic account information.
-    fn basic(&mut self, address: H160) -> AccountInfo;
+    fn basic(&mut self, address: H160) -> Result<Option<AccountInfo>, &'static str>;
```

Comment:
would it be possible to make this a `Cow`, this way we could include additional context in the error like address etc

Reply:
Adding associate type Error to database trait would maybe make this better

Reply:
You can now specify `Error` type inside `Database`.